### ISSUE TITLE
this

### ISSUE THREAD

**Issue by kreinholz** [susanBuck/e28-fall19#135]
I’m one of those JavaScript developers who always think I know what “this” means, only to find a new edge case that throws me off.

In the case of Vue and using “this” to access variables defined within data, I came across the issue of function scope denying access to the “this” I wanted. The answer was surprisingly simple, but as it’s explained more eloquently by another, I’m posting a link that I hope will help others out if you get stuck in the same rut I was in:

<https://www.reddit.com/r/vuejs/comments/5ae7fj/how_do_i_reference_data_from_a_function_inside_a>

Bottom line, a single line of code outside your loop/function defining a pointer to “this” can get you access to your data properties even from a function within a function in one of your methods.

**Comment by gaderrick** [comment]
Thanks for "this" (pun) post... I too had to reteach myself what seemed like it should just be working. I like his explanation.

**Comment by extra808** [comment]
This is covered in the [MDN documentation of setInterval()](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setInterval#The_this_problem), "Code executed by setInterval() runs in a separate execution context than the function from which it was called. As a consequence, the this keyword for the called function is set to the window (or global) object..."

[setTimeout()](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout#The_this_problem) has the same issue.

**Comment by susanBuck** [comment]
Thanks for sharing @kreinholz! 

Another approach is to use arrow functions; because they don't have their own execution context the `this` is inherited.

Example for reference - [ProductPage.vue#L51](https://github.com/susanBuck/e28/blob/master/zipfoods/src/components/pages/ProductPage.vue#L51)

```js
setTimeout(() => (this.addAlert = false), 2000);
```

**Comment by ihimawan** [comment]
This is definitely helpful, thanks for sharing !
